js_modules/dagster-ui/packages/ui-core/src/launchpad/LaunchpadSession.tsx

```diff
@@ -299,24 +294,15 @@ const LaunchpadSession = (props: LaunchpadSessionProps) => {
   const onScaffoldMissingConfig = () => {
     const config = runConfigSchema ? scaffoldPipelineConfig(runConfigSchema) : {};
     try {
-      const runConfigData = yaml.parse(sanitizeConfigYamlString(currentSession.runConfigYaml));
-      const updatedRunConfigData = merge(config, runConfigData);
-      const runConfigYaml = yaml.stringify(updatedRunConfigData);
-      onSaveSession({runConfigYaml});
+      onSaveSession({runConfigYaml: mergeYaml(config, currentSession.runConfigYaml)});
     } catch (err) {
       showCustomAlert({title: 'Invalid YAML', body: YAML_SYNTAX_INVALID});
     }
   };
 
   const onExpandDefaults = () => {
     if (rootDefaultYaml) {
-      const defaultsYaml = yaml.parse(sanitizeConfigYamlString(rootDefaultYaml));
-
-      const currentUserConfig = yaml.parse(sanitizeConfigYamlString(currentSession.runConfigYaml));
-      const updatedRunConfigData = merge(defaultsYaml, currentUserConfig);
-      const mergedYaml = yaml.stringify(updatedRunConfigData);
-
-      onSaveSession({runConfigYaml: mergedYaml});
+      onSaveSession({runConfigYaml: mergeYaml(rootDefaultYaml, currentSession.runConfigYaml)});
```

Comment:
onSaveSession((currentSession) => ({runConfigYaml: mergeYaml(rootDefaultYaml, currentSession.runConfigYaml)}));

Reply:
Hmm it looks like this should work (onSaveSession is a `useSetStateUpdateCallback` function reference, and that mimics SetStateAction), but we don't do it anywhere in the LaunchpadSession currently.  Going to do do this in a small follow-up PR and change all the callsites at once

Reply:
edit: it looks like we're in a weird spot with this because onSaveSession takes a partial state (our previous solution to ensure it didn't get stale values), which means that the type on the current state is also the partial state, not the full state  🤔

Reply:
Can't we just set the full state and any fields we want to get rid of we set as undefined/null ? I do agree this should just take the full state though, we can update callsites to pass in the full state and they can preserve the old behavior by doing `onSaveSession({...currentSession, {rootConfigYaml: ...}})`

Reply:
Yeah I think we can -- we just need to read/write the full state using that spread operator everywhere. It actually looks like the function-setter won't work until we do that because the "current value" is just the last partial that was set currently. Was trying to see if I could change that, but just setting the entire state each time would be better. 

Going to do this in a follow-up tonight, it's more lines of code than I expected!